homeassistant/components/google_assistant/trait.py

```diff
@@ -520,3 +525,123 @@ def query_attributes(self):
                     climate.ATTR_OPERATION_MODE:
                         self.google_to_hass[params['thermostatMode']],
                 }, blocking=True)
+
+
+@register_trait
+class StartStopTrait(_Trait):
+    """Trait to offer start, stop, and pause functionality.
+
+    https://developers.google.com/actions/smarthome/traits/startstop
+    """
+
+    name = TRAIT_STARTSTOP
+    commands = [
+        COMMAND_STARTSTOP,
+        COMMAND_PAUSEUNPAUSE
+    ]
+
+    @staticmethod
+    def supported(domain, features):
+        """Test if state is supported."""
+        if domain != vacuum.DOMAIN:
+            return False
+
```

Comment:
Probably a good idea to check for SUPPORTS_STATE here.  That will exclude any platforms that haven't been ported yet